models/dataset_test.go

```diff
@@ -0,0 +1,175 @@
+package models
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/ONSdigital/go-ns/log"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestCreateDataset(t *testing.T) {
+	t.Parallel()
+	Convey("Successfully return without any errors", t, func() {
+		Convey("when the dataset has all fields", func() {
+			b, err := json.Marshal(inputDataset)
+			if err != nil {
+				log.ErrorC("Failed to marshal test data into bytes", err, nil)
+				os.Exit(1)
+			}
+			r := bytes.NewReader(b)
+			dataset, err := CreateDataset(r)
+			So(err, ShouldBeNil)
+			So(dataset.CollectionID, ShouldEqual, "12345678")
```

Comment:
I know `git mv ...` would have made you look less guilty of this, but I would not have duplicated/hardcoded the values, here, but re-used the test-data, e.g.:
`So(dataset.CollectionID, ShouldEqual, inputDataset.CollectionID)`